src/data/polynomial/mirror.lean

```diff
@@ -183,4 +184,124 @@ end
 
 end mirror
 
+section norm2
+
+variables {R : Type*} [comm_ring R] (p q : polynomial R)
+
+noncomputable def norm2 : quadratic_form R (polynomial R) :=
+finsupp.norm_sq
+
+lemma norm2_apply : p.norm2 = p.sum (λ k c, c ^ 2) :=
+finsupp.norm_sq_apply p
+
+lemma norm2_eq_sum_of_support {s : finset ℕ}
+  (h : p.support ⊆ s) : p.norm2 = s.sum (λ k, (p.coeff k) ^ 2) :=
+p.norm2_apply.trans (finset.sum_subset h (λ k h1 h2, by
+{ rw [finsupp.not_mem_support_iff.mp h2], exact zero_pow zero_lt_two }))
+
+lemma norm2_monomial (k : ℕ) (a : R) : (monomial k a).norm2 = a ^ 2 :=
```

Comment:
Can you make the corresponding lemma for `finsupp.lsingle` (`monomial` is equal to this, by definition) in `quadratic_form`?